### PR TITLE
Issue 3937 notification email

### LINKED ISSUE #3937: Build Notification Email
### Definition of Done
- [ ] Use current Gitcoin email template
- [ ] The name of the person should be **dynamic** based on who it's sent to.
- [ ] Build a simple email that sends users a **weekly** reminder of notifications that they need to read in Gitcoin. Schedule the weekly trigger for all users
- [ ] Use cron
- [ ] The backend endpoint is located at `/inbox/notifications/`
https://github.com/gitcoinco/web/blob/master/scripts/crontab (edited) 
- [ ] Clicking on the button takes the user to the notification page.
- [ ] Email must be responsive.
- [ ] All tests must pass before code is merged
- [ ] Review with Gitcoin core team

### Reference

screen shot 2019-03-08 at 2 05 24 pm

[Notification Email 2.zip](https://github.com/gitcoinco/web/files/2947118/Notification.Email.2.zip)

### PR DESCRIPTION
**Pull request by iamonuwa**
##### Description

Send weekly unread notification emails

##### Checklist

- [x] Read and conforms to the [contributor guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Read and conforms to the [style guidelines](https://docs.gitcoin.co/mk_styleguide/)
- [x] Linter status: 100% pass
- [x] Added relevant tests
- [ ] Tests pass and code coverage has not decreased
- [x] Changes don't break existing behavior
- [x] Commit message follows [commit guidelines](https://docs.gitcoin.co/mk_contributors/#step-4-commit)
- [x] [Code documentation](https://docs.gitcoin.co/mk_contributors/#docstrings)
- [x] Completed: Code review by core team and any requested changes

##### Refers/Fixes
Fixes #3937 

###### Contributor

- [x] Read and followed the [Contributor Guidelines](https://docs.gitcoin.co/mk_contributors/)
- [x] Tested all changes **locally**
- [x] Verified existing functionality
- [x] Ran `make test` and everything passed!

###### Reviewer

- [ ] Affirm contributor guidelines have been followed and requested changes made
- [ ] CI tests and linting pass
- [ ] No conflicts (migrations, files, etc)
- [ ] Regression tested against staging or local deployment

###### Funder

- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

**Comment by thelostone-mc** [comment]
@iamonuwa i see tests \m/ 
btw could you throw in a screenshot of the mail too ?

**Comment by iamonuwa** [comment]
> @iamonuwa i see tests \m/
> btw could you throw in a screenshot of the mail too ?

Will send it in tomorrow, am AFK at the moment.

**Comment by PixelantDesign** [comment]
@thelostone-mc let me know when good to pay out!

**Comment by iamonuwa** [comment]
image

**Comment by iamonuwa** [comment]
cc: @thelostone-mc

File: app/marketing/mails.py [hunk 1]
```diff
@@ -686,6 +685,33 @@ def weekly_recap(to_emails=None):
         finally:
             translation.activate(cur_language)
 
+def unread_notification_email_weekly_roundup(to_emails=None):
+    if to_emails is None:
+        to_emails = []
+    
+    for to_email in to_emails:
+        cur_language = translation.get_language()
```
Comment (rafalkowalski) [inline app/marketing/mails.py]: Could you execute translation.get_language() before loop?

File: app/marketing/mails.py [hunk 2]
```diff
@@ -686,6 +685,33 @@ def weekly_recap(to_emails=None):
...
+            from_email = settings.PERSONAL_CONTACT_EMAIL
+
+            if not should_suppress_notification_email(to_email, 'weeklyrecap'):
+                send_mail(
+                    from_email, 
+                    to_email, 
+                    subject, 
+                    text, 
+                    html, 
+                    from_name="Kevin Owocki (Gitcoin.co)",
```
Comment (rafalkowalski) [inline app/marketing/mails.py]:
```
from_name="Kevin Owocki (Gitcoin.co)"
categories=['marketing', func_name()]
```

Above value are repeated 3 times could you create const variables at the top of file.

File: app/marketing/management/commands/send_unread_notification_emails_roundup.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+from marketing.models import EmailSubscriber
+
+warnings.filterwarnings("ignore", category=DeprecationWarning)
+
+
+check_already_sent = False
+
+
+def is_already_sent_this_week(email):
+    from marketing.models import EmailEvent
```
Comment (rafalkowalski) [inline app/marketing/management/commands/send_unread_notification_emails_roundup.py]: Could you move imports to the top?

File: app/marketing/management/commands/send_unread_notification_emails_roundup.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    from django.utils import timezone
+    then = timezone.now() - timezone.timedelta(hours=12)
+    QS = EmailEvent.objects.filter(created_on__gt=then)
```
Comment (rafalkowalski) [inline app/marketing/management/commands/send_unread_notification_emails_roundup.py]: QS variable name isn't readable. Why is this variables uppercase?

File: app/marketing/management/commands/send_unread_notification_emails_roundup.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        start_counter = options['start_counter']
+
+        queryset = EmailSubscriber.objects.all()
+        if exclude_startswith:
+            queryset = queryset.exclude(email__startswith=exclude_startswith)
+        if filter_startswith:
+            queryset = queryset.filter(email__startswith=filter_startswith)
+        queryset = queryset.order_by('email')
+        email_list = list(set(queryset.values_list('email', flat=True)))
+        # list.sort(email_list)
```
Comment (rafalkowalski) [inline app/marketing/management/commands/send_unread_notification_emails_roundup.py]: Please remove commented code

File: app/marketing/management/commands/send_unread_notification_emails_roundup.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+
+        print("got {} emails".format(len(email_list)))
+
+        counter = 0
+        for to_email in email_list:
```
Comment (rafalkowalski) [inline app/marketing/management/commands/send_unread_notification_emails_roundup.py]: Instead of counter you could use enumerate function.

**Comment by thelostone-mc** [comment]
@iamonuwa mind addressing the comments left by @rafalkowalski + rebase with master ?  ^_^
@PixelantDesign https://github.com/gitcoinco/web/pull/3959#issuecomment-472767747 look good ?

**Comment by iamonuwa** [comment]
> @iamonuwa mind addressing the comments left by @rafalkowalski + rebase with master ? ^_^
> @PixelantDesign [#3959 (comment)](https://github.com/gitcoinco/web/pull/3959#issuecomment-472767747) look good ?

Yes, will make changes to reflect the review

File: app/marketing/mails.py (hunk 2 above)
Comment (iamonuwa) [reply app/marketing/mails.py]: Can you elaborate on this please

File: app/marketing/management/commands/send_unread_notification_emails_roundup.py (hunk 6 above)
Comment (iamonuwa) [reply app/marketing/management/commands/send_unread_notification_emails_roundup.py]: As well as this, why enum over counter

Comment (thelostone-mc) [reply app/marketing/management/commands/send_unread_notification_emails_roundup.py]: I believe @rafalkowalski meant this ->`for index, to_email in enumerate(email_list):`

That way you wouldn't have to create a counter variable and increment it as the index is already available

File: app/marketing/mails.py (hunk 2 above)
Comment (thelostone-mc) [reply app/marketing/mails.py]: He means the two values are used in multiple places , so he's suggesting we move it a constant and just reuse the constant.
I'm good either way so I leave it to you @iamonuwa to make the call

Comment (iamonuwa) [reply app/marketing/mails.py]: Am fine with leaving as it is.

File: app/marketing/management/commands/send_unread_notification_emails_roundup.py (hunk 6 above)
Comment (iamonuwa) [reply app/marketing/management/commands/send_unread_notification_emails_roundup.py]: I think this is cleaner and better

**Comment by thelostone-mc** [comment]
@iamonuwa we've got CI failures ! mind fixing them up ?

**Comment by iamonuwa** [comment]
> @iamonuwa we've got CI failures ! mind fixing them up ?

Aye captain!

**Comment by thelostone-mc** [comment]
@iamonuwa lemme know when CI is green ^_^

**Comment by iamonuwa** [comment]
> @iamonuwa lemme know when CI is green ^_^

Working on it

**Comment by iamonuwa** [comment]
cc: @thelostone-mc

File: app/retail/emails.py [hunk 7]
```diff
@@ -1050,6 +1073,11 @@ def weekly_recap(request):
     return HttpResponse(response_html)
 
 
+@staff_member_required
+def unread_notification_email_weekly_roundup(request):
+    response_html, _ = render_unread_notification_email_weekly_roundup('mark.beacom@consensys.net')
+    return HttpResponse(response_html)
+
```
Comment (thelostone-mc) [inline app/retail/emails.py]: we need an extra empty line (keeping it consistent )

**Pull request merged**